ads/firstimpression.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {validateData, writeScript} from '../3p/3p';
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function firstimpression(global, data) {
+  validateData(data, ['zoneId', 'websiteId']);
+
+  const getURLHashParameter_ = function (name) {
```

Comment:
can we use https://github.com/ampproject/amphtml/blob/c1aedd0050e32e58b7851832aa2a2b0561d8b6df/src/url.js#L119 here?

Reply:
No, because I need to return value of a specific parameter that appears in URL fragment, not the entire Location object

Reply:
I can, actually, change this to use `URLSearchParams` if it is better. Code will probably be a little longer.

Reply:
actually I think we already give you `data.location`. You can do `location.hash`?

Reply:
are these hash params only used for testing? I also have some concern that pubs can use hash for links that might break all of this?

Reply:
`data` only contains tag attributes that begin with `data-`, and in addition `ampSlotIndex`.

Reply:
`global.context.location` Again sorry for the confusion.

Reply:
Done